### PR TITLE
lazy box and normal box support

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart [hunk 1]
```diff
@@ -340,7 +351,9 @@ class CommitLogKeyStore implements LogKeyStore<int, CommitEntry?> {
     var keys = _getBox().keys;
     var value;
     await Future.forEach(keys, (key) async {
-      value = await _getBox().get(key);
+      value = await (_isLazy
+          ? (_getBox() as LazyBox).get(key)
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: Do we have to cast to call a get()? it should be polymorphic.

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart [hunk 2]
```diff
@@ -49,7 +55,9 @@ class CommitLogKeyStore implements LogKeyStore<int, CommitEntry?> {
   @override
   Future<CommitEntry?> get(int commitId) async {
     try {
-      var commitEntry = await _getBox().get(commitId);
+      var commitEntry = await (_isLazy
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: Do we have to cast to call a get()? it should be polymorphic.

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart [hunk 3]
```diff
@@ -31,7 +33,11 @@ class CommitLogKeyStore implements LogKeyStore<int, CommitEntry?> {
     }
 
     this.storagePath = storagePath;
-    await Hive.openBox(_boxName);
+    if (_isLazy) {
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: Let's have a openBox() ideally in the base class for all stores that take isLazy and other params that initializes properly. Similarly, getBox() also needs to be in a base class.

Another option is, like java if Dart supports default methods have openBox and getBox() in an interface and make these classes implement or extend it.

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart (hunk 1 above)
Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: get method is not available in the base class(BoxBase). it is available only in the sub class (Box and LazyBox)

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart (hunk 3 above)
Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: sure..will make this change

File: at_secondary/at_persistence_secondary_server/lib/src/log/accesslog/access_log_keystore.dart [hunk 4]
```diff
@@ -48,7 +51,9 @@ class AccessLogKeyStore implements LogKeyStore<int, AccessLogEntry?> {
   @override
   Future<AccessLogEntry?> get(int key) async {
     try {
-      var accessLogEntry = await _getBox().get(key);
+      var accessLogEntry = await (super.isLazy
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/log/accesslog/access_log_keystore.dart]: Should this be super.getBox).get()?

Comment (VJag) [reply at_secondary/at_persistence_secondary_server/lib/src/log/accesslog/access_log_keystore.dart]: get(key) is available on both the boxes. Do we need a cast()?

**Review by VJag — APPROVED** [review]

**Pull request merged**